python/hail/tests.py

```diff
@@ -769,3 +770,43 @@ def test_concordance(self):
 
         glob, samples, variants = bn1.concordance(bn2)
         self.assertEqual(samples.sample_annotations[samples.sample_ids[0]].concordance, glob)
+
+    def test_hdfs_files(self):
+        data = ['foo', 'bar', 'baz']
+        data.extend(map(str, range(100)))
+
+        with hdfs_write('/tmp/test_out.txt') as f:
+            for d in data:
+                f.write(d)
+
+        with hdfs_read('/tmp/test_out.txt') as f:
+            data2 = [line for line in f]
+
+        self.assertEqual(data, data2)
+
+        with hdfs_write('/tmp/test_out.txt.gz') as f:
+            for d in data:
+                f.write(d)
+
+        with hdfs_read('/tmp/test_out.txt.gz') as f:
+            data3 = [line for line in f]
+
+        self.assertEqual(data, data3)
+
+        hdfs_copy('/tmp/test_out.txt.gz', '/tmp/test_out.copy.txt.gz')
+
+        with hdfs_read('/tmp/test_out.copy.txt.gz') as f:
+            data4 = [line for line in f]
+
+        self.assertEqual(data, data4)
+
+        with open('/tmp/numbers.txt', 'w') as f:
```

Comment:
We can't do this, I think. if hdfs is the default file system, then the hdfs_read won't see it